### ISSUE TITLE
Convert Gal Entities to Liters

### ISSUE THREAD

**Issue by loictinant** [rewardone/homeassistant-culligan-water-softener#13]
Hello,

I would like to convert Gal Entities to Liters, but how do I do it?

When I click on an entity to put in liters, I validate but everything remains in Gal?

How do I do it?

**Comment by rewardone** [comment]
Great question. I'm not sure why the UI configuration doesn't work. 

An alternative is to customize entities via yaml. It's not ideal, but I'm unsure why the UI changes aren't working. 

In Configuration.yaml:
`homeassistant:
  customize: !include customize.yaml`

In customize.yaml:
`sensor.DSN_average_daily_usage:
  unit_of_measurement: L:`

Restart Home Assistant. You may also need to purge the recorder/history stats for the entity, which will have values originating in gal. 

While this forcibly changes the unit, I am unsure if it will convert them. After I changed my entity, for example, my sensor stayed at the same value, which was in gallons. Keep an eye on your entity value if you perform this customization. 

The further less ideal state would be to create separate helper entities that have the correct conversions.. You can hide the Culligan entities.

**Comment by rewardone** [comment]
According to a [post from 2021](https://community.home-assistant.io/t/list-of-valid-unit-of-measurement-options/299209/9)

`Only c and f get converted. Everything else remains untouched`

unit_of_measurement is simply a string value, so no conversions can be expected. If the data you're receiving from Culligan is in L, you'll just need to change the unit_of_measurement. If the data you're receiving from Culligan is in gal, you'll need to create a helper, it seems, to have this value in L.

**Comment by loictinant** [comment]
Bonjour, 

Alors l'unité change bien, mais la conversion n'est pas du tout la bonne.

Exemple : J'ai sur "average daily water usage" 83 gallons, et en Litres j'ai 23 L.
Mais en réalité, 83 gallons fait 314 Litres

**Comment by elmincho80** [comment]
Indeed, we're in France and the unit of measurement is liters... The sensors you're looking for are in gal :( so it's not at all meaningful to us French.

Any idea if other sensors are in liters?

Thanks in advance.

**Comment by rewardone** [comment]
I would need someone to provide the properties provided on the object. If Culligan provides a "total_liters_today" sensor or other liter specific properties. I don't think every property is printed in debug mode because there are so many. You can see the properties I know about in [const](https://github.com/rewardone/homeassistant-culligan-water-softener/blob/main/custom_components/culligan/const.py#L46). 

My current recommendation would be to create template helper entities based off the gallon entities to do the proper conversion. The gallon entities could then be hidden.

While the numeric conversion would be pretty simple in the integration, it would probably take me some troubleshooting to add in a UI binary selector to base entity creation/conversion on automatically.

**Comment by elmincho80** [comment]
Many tanks for your answer :) Indeed I used a sensor template to convert gallons to liters and it works perfectly. I can even use your sensor with a lovelace with on-the-fly conversion.
